cirq-core/cirq/protocols/act_on_protocol.py

```diff
@@ -11,24 +11,28 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-"""A protocol that wouldn't exist if python had __rimul__."""
 
-from typing import Any, TYPE_CHECKING, Union
+from typing import TYPE_CHECKING, Union, Sequence, Any
 
 from typing_extensions import Protocol
 
+from cirq import ops
 from cirq._doc import doc_private
 from cirq.type_workarounds import NotImplementedType
 
 if TYPE_CHECKING:
-    pass
+    import cirq
 
 
 class SupportsActOn(Protocol):
     """An object that explicitly specifies how to act on simulator states."""
 
     @doc_private
-    def _act_on_(self, args: Any) -> Union[NotImplementedType, bool]:
+    def _act_on_(
+        self,
```

Comment:
I think we still need a distinct Supports class for the two different interfaces.